packages/transaction-controller/src/utils.ts

```diff
@@ -226,8 +241,122 @@ export function transactionMatchesNetwork(
   if (transaction.chainId) {
     return transaction.chainId === chainId;
   }
+
   if (transaction.networkID) {
     return transaction.networkID === networkId;
   }
   return false;
 }
+
+/**
+ * Determines the type of the transaction by analyzing the txParams.
+ * It will never return TRANSACTION_TYPE_CANCEL or TRANSACTION_TYPE_RETRY as these
+ * represent specific events that we control from the extension and are added manually
+ * at transaction creation.
+ *
+ * @param txParams - Parameters for the transaction.
+ * @param ethQuery - EthQuery instance.
+ * @returns A object with the transaction type and the contract code response in Hex.
+ */
+export async function determineTransactionType(
+  txParams: Transaction,
+  ethQuery: EthQuery,
+): Promise<InferTransactionTypeResult> {
+  let result;
+  let contractCode;
+  const { data, to } = txParams;
+  const name = parseStandardTokenTransactionData(data)?.name;
+
+  if (data && !to) {
```

Comment:
Rather than all the nesting, would it be cleaner if we early return for this condition, plus on line 293?

Reply:
Make sense, refactored `determineTransactionType`.